doc/release-notes/whats-new.en.rst

```diff
@@ -58,4 +58,6 @@ using cpp:func:`Dbg` and related functions, as in |TS| core code.
 C++ Plugin API Deprecated
 ^^^^^^^^^^^^^^^^^^^^^^^^^
 
-It is deprecated in this release.  It will be deleted in ATS 11.
+It is deprecated in this release.  It will be deleted in ATS 11.  The ``noncopyable.h``
+header, with the ``noncopyable`` class, is deleted in this release.  Use the new use of the
```

Comment:
I proposed on the dev mailing list to remove it.  No one asked for it to be kept.

Reply:
I said remove it when we delete the C++ API. Not when we deprecate the C++ API. If your intention was to remove it now, I'll throw -1 on the list.
https://lists.apache.org/thread/fmhdqlfl3zn5lvgx4nyqr09fqjvj7y3c

Reply:
Breaking changes are allowed for major releases.  We haven' t followed the run in the past of deprecating first before any breaking change.  This isn't a major change in my opinion.

Reply:
> Breaking changes are allowed for major releases.

I know.

> We haven' t followed the rule in the past of deprecating first before any breaking change. 

The rule exists, and I think we follow it in general with some exceptions. If you think the rule has been ignored and the rule does not make sense you should probably propose removing the rule first.

What is the point of deprecating C++ API if it's broken?
Why do we need to remove noncopyable.h right now?

Reply:
Well would be nice if the rules were written down somewhere. But, to the best of my knowledge, breaking changes are permitted in a major release, with no strict requirement for deprecation. We have done deprecation first for major changes, but I don't know of any rules as to what is a major change. As they say, why put off till tomorrow what you can do today. That is, procrastination bad. Note that the compress plugin is using noncopyable, even though it does not use anything else in the C++ API. If we leave it around for another year plus, til release 11, that's more opportunity for it to be used. Kill one roach now, or a thousand later.

Reply:
> why put off till tomorrow what you can do today.

Because breaking compatibility prevents smooth migration from ATS 9 to 10.

> If we leave it around for another year plus, til release 11, that's more opportunity for it to be used.

No. We deprecate things to prevent that.

Reply:
We didn't deprecate use of straight C in plugins in 9, before we blocked it in 10.  That's a bigger issue in migration than removing noncopyable is.

Reply:
I think we should remove this as part of moving to a C++ API for plugins. That's a far more major change because it deprecates plugins written purely in C (and those exist, I know Yahoo! has at least one). Its existence is potentially confusing, in that it should not be used in any circumstances. I also think it's acceptable because no functionality has been removed - anyone using this can trivially replace it using language mechanisms.

Reply:
> We didn't deprecate use of straight C in plugins in 9, before we blocked it in 10.

What plugin code change did it require?

> Its existence is potentially confusing, in that it should not be used in any circumstances.

Anything deprecated would be "confusing". Are you suggesting not to deprecate anything?

>  I also think it's acceptable because no functionality has been removed

I don't think that matters. Could we add a parameter to an API function if no functionality is removed? I'd say no.

Reply:
> > We didn't deprecate use of straight C in plugins in 9, before we blocked it in 10.
> 
> What plugin code change did it require?

Look at the chances to the tslua core plugin:  https://github.com/apache/trafficserver/pull/10007/files

Reply:
I'm not sure which changes were truly needed for the migration, but looks much trivial. NULL works with C++, right? Even if some of the changes were actually needed, I don't get why we should put extra work to unnecessarily remove noncopyable soon.